### PR TITLE
chore: update issue and PR templates

### LINKED ISSUE #3495: [Proposal] GitHub Usage and Process
As previously discussed, I've been taking a look at our issues, milestones, labels, projects, etc. and have put together a set of recommendations.

# GitHub Specifics

## Milestones

Our current set of milestones are out of date. My recommendation is to close out all of our existing milestones. Going forward, we can center milestones around particular major features or innovations. Some milestones might be "Router v1", "New Provider Infrastructure", "Fluent UI Refactor", etc.

## Labels

I took a look at our existing set of labels, as well as the labels used by Aurelia and .NET, to get a survey of different patterns being used. I like some of what @awentzel put in place for our labels and think we can benefit from making further use of what's there. I've pulled some ideas from Aurelia and .NET that have the notion of "topics" or "areas" as well. Here's what I've come up with by doing a bit of cleanup/consistency work on our own labels and combining that with the area idea.

### Issue Types

Each of these labels will have a distinct color and is based off of our current conventional changelog git commit types. Every issue should receive exactly one of these labels as part of the triage process.

- `bug`
- `feature`
- `improvement`
- `chore`
- `docs`
- `rfc`
- `spec`
- `investigation`

### Issue Status

All of these labels have the same color and represent the status of the issue. Every issue should receive at least one of these labels as part of the triage process. The `status:triage` status is special. Each newly created issue should start out in this state but should not remain in this state after the triage process is finished.

- `status:triage`
- `status:needs-information`
- `status:needs-investigation`
- `status:blocked`
- `status:under-review`
- `status:planned`
- `status:in-progress`

### Issue Areas

All of these labels have the same color and represent the area of the codebase or conceptual feature-set that the issue is related to. As part of the triage process, an issue should be given at least one area label.

- `area:fast-element`
- `area:fast-foundation`
- `area:fast-components`
- `area:fast-components-msft`
- `area:tools`
- `area:dev-ops`
- `area:cloud`

### Community Labels

All of thees labels should have the same color and can optionally be added at any point prior to closure to signal the community of special opportunities to contribute.

- `community:good-first-issue`
- `community:help-wanted`

### Issue Closure

All of these labels have the same color and represent the reason why the issue was closed. When an issue is closed, its status label(s) must be removed and at least one close label should be added.

- `closed:done`
- `closed:by-design`
- `closed:duplicate`
- `closed:no-repro`
- `closed:obsolete`
- `closed:stale`

## Projects

I recommend that we begin using GitHub projects more to track our backlog priority and in-progress work. To do this, I think we can just clean up the projects we have a little bit and have something we can work with. I recommend we go with the following set of initial projects:

- Architecture - Track core architecture and capabilities work primarily in `fast-element` and `fast-foundation`
- Components - Track work on new components or fixes/improvements to existing components. This is likely to span across `fast-foundation` as well as the design system packages.
- Design Systems - Tracks core work on design systems in `fast-foundation` as well as design-system-specific work in their packages.
- Tooling - Tracks work on our core tools as well as 3rd party tool plugins (e.g. Figma)
- Documentation - Tracks work on the documentation.
- Plugins - Tracks work on optional packages such as a future router, dependency injection, etc.

## Issue Templates

We can create a few more issue template options, aligning with the basic issue types listed above.

- bug
- feature or impreovement
- chore
- docs
- rfc
- spec
- investigation

## Pull Request Templates

Right now we have a single PR template that is used. We can split this out into multiple templates, which would make certain scenarios such as spec PRs or component PRs more clear. The following templates are my recommendation:

- bug
- feature or improvement
- component
- chore
- docs
- spec

# Process Summary

At the highest level, our team strategic planning meetings will identify our milestones. We may also choose to create special project boards, depending on the size or priority of the work.

Issues will either be created by us, in which case they can have labels immediately assigned and be prioritized in any related boards.

Or the issues will be created by the community. Each morning someone can triage the new issues. GitHub will add the `status:triage` label for us, so it will be easy to locate issues that need to be triaged. Once an issue is triaged, the person responsible should note if there are any issues that require immediate action. Examples are:

* needs information - follow up briefly with the issue reporter to get information
* security/privacy/accessibility issue - follow up with lead to discuss and determine if current work should be interrupted to address

If the issue does not represent an immediate need, then the team member should make sure to put an appropriate status. Anything tagged with `status:under-review`, `status:planned`, or `status:blocked` can be prioritized as part of the team strategy meeting.

We can rotate issue triage across the team, by assigning a particular day to each person(s). (I'm happy to do the first several rounds myself in order to get existing issues cleaned up.)

Each sprint we'll pull issues from the ordered backlog, assign them to team members and mark them with `status:in-progress`. When the issue is resolved, it can be tagged with a `closed` label.

## Open Questions

* Do we want to do estimation as part of the backlog grooming process? If so, what is the best way to do that on GitHub? Should we just add a comment with the estimate? Should we have a set of labels based on an abstraction (xs/s/m/l/xl or fibonacci days .5,1,2,3,5,8,13)
* If we do estimation in GitHub, is there an easy way to measure our agility so that we can improve the accuracy of our sprint commitments over time?

### PR DESCRIPTION
**Pull request by EisenbergEffect**
# Pull Request

## 📖 Description

Updating our issue and PR templates is the last piece of the work to refresh our GitHub usage approach. We'd like to have a friendly set of templates and guidance for the most common scenarios. This is my take on that based on some prior projects of mine as well as notes I've made over the last few months from team feedback.

Note: While it's technically possible to have multipe PR tempaltes, you can't select them through the UI. The submitter needs to know the template name and add it to the query string, which isn't very user friendly. For this reason, this PR sticks with a single PR template and instead re-organizes the checklists.

### 🎫 Issues

* Closes #3259 
* Closes #3495

## 👩‍💻 Reviewer Notes

This PR (manually) uses the new template, so take a moment to think through how you are experiencing the template right now. A primary characteristic of a good PR is to present an excellent reading experience since the PR will be read by many people, not only now but possibly months or years later. After considering this, have a look at the changed files in this PR. They are all markdown as this is a content-only change. No code is affected. The new templates make use of emojis and commented guidance for each section. What do you think of this approach?

## 📑 Test Plan

No testing was done as part of this PR other than viewing how files look on GitHub.

### General

- [x] I have followed the title guidelines for my PR
- [ ] I have included a change request file using `$ yarn change` - NOTE: Note implemented yet. Should we leave this off?
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

## ⏭ Next Steps

This completes the known initial set of work to update our GitHub use for FAST. We'll continue to take feedback and make modification to our templates, labels, issues, PRs, projects, milestones, etc.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.md
Comment (chrisdholt) [inline .github/ISSUE_TEMPLATE/bug-report.md:1]: What do you think of asking if they're willing to contribute a fix?

Comment (EisenbergEffect) [reply .github/ISSUE_TEMPLATE/bug-report.md:1]: Sounds great. I'll add it under the possible solution section.

File: .github/ISSUE_TEMPLATE/bug-report.md
Comment (nicholasrice) [inline .github/ISSUE_TEMPLATE/bug-report.md:38]: Thoughts on moving this just below "Bug Report"? To me, the repro steps are super important and set context for "expected behavior" and "current behavior"

Comment (EisenbergEffect) [reply .github/ISSUE_TEMPLATE/bug-report.md:38]: Yeah, makes sense. Give me a sec...

**Review by nicholasrice — APPROVED** [review]
Looks good! Just one thought above.

**Review by janechu — APPROVED** [review]

File: .github/pull_request_template.md
Comment (chrisdholt) [inline .github/pull_request_template.md:47]: Assuming we are waiting for the updated publishing work to go in for this? If that is the case, then we don't need the title guidelines. You can title your PR anything once we shift to the `yarn change` process.

File: .github/pull_request_template.md
Comment (chrisdholt) [inline .github/pull_request_template.md:74]: Per above, I don't think we need this in the world of `yarn change`

**Review by awentzel — COMMENTED** [review]
Do we need the emoji's in the headings?  Do they add value?  To my eyes, they are a bit distracting and moves my focus away from the text.

File: .github/ISSUE_TEMPLATE/feature-request.md
Comment (awentzel) [inline .github/ISSUE_TEMPLATE/feature-request.md:9]: Is there a way to templatize the duplicative content that used across templates?

File: .github/ISSUE_TEMPLATE/feature-request.md
Comment (awentzel) [inline .github/ISSUE_TEMPLATE/feature-request.md:28]: Perhaps, "What implementation or solution would be ideal for you?" instead of both sentences that ring fairly similar.

File: .github/ISSUE_TEMPLATE/feature-request.md
Comment (awentzel) [inline .github/ISSUE_TEMPLATE/feature-request.md:33]: It may be better to break the second sentence into a new line to be consistent and make it more scannable, thereby readable and more likely to get the requested details.

**Review by awentzel — APPROVED** [review]

File: .github/pull_request_template.md
Comment (EisenbergEffect) [reply .github/pull_request_template.md:47]: We can wait until we make the updates. It was an open question in my mind, but let's hold this for the beachball change and bring it all in at the same time. I'll remove the title guideline-related items.

File: .github/ISSUE_TEMPLATE/feature-request.md
Comment (EisenbergEffect) [reply .github/ISSUE_TEMPLATE/feature-request.md:9]: Not that I'm aware of. I read all the docs on issue and PR templates and nothing comes up for that.

**Comment by EisenbergEffect** [comment]
This is ready to go but I'm marking this as `status:blocked` for now. We'll unblock and merge once the new beachball release setup in in place.

**Comment by janechu** [comment]
Removed stale warning as this is temporarily blocked, not stale.

File: .github/ISSUE_TEMPLATE/documentation.md
Comment (awentzel) [inline .github/ISSUE_TEMPLATE/documentation.md:3]: 'Find inaccuracies in the documentation or something missing?'

**Comment by awentzel** [comment]
@EisenbergEffect per your question regarding `yarn change`.  This feature is in master, so you can include in the PR.  We won't use it until we have finalized beachball implementation (it's been tested locally, just not in production scenarios. We need the production scenarios to be fired off manually by @chrisdholt, once that passes I have a quick edit to the CI and it's should run nightly from that point going forward.

**Review by eljefe223 — APPROVED** [review]

**Comment by chrisdholt** [comment]
@EisenbergEffect I think we are ready to bring this in :)

**Review by chrisdholt — APPROVED** [review]

**Comment by EisenbergEffect** [comment]
@chrisdholt Should I merge now or wait to do it as part of our meeting tomorrow?

**Comment by chrisdholt** [comment]
> @chrisdholt Should I merge now or wait to do it as part of our meeting tomorrow?

I think you're probably good now to be honest. We can do it tomorrow as well, but I don't see a huge problem with merging prior to tomorrow.

**Pull request merged**
